src/hotspot/share/opto/block.cpp

```diff
@@ -37,11 +37,8 @@
 #include "utilities/copy.hpp"
 #include "utilities/powerOfTwo.hpp"
 
-void Block_Array::grow( uint i ) {
-  _nesting.check(_arena); // Check if a potential reallocation in the arena is safe
-  if (i < Max()) {
-    return; // No need to grow
-  }
+void Block_Array::grow(uint i) {
+  assert(i >= Max(), "must be an overflow");
```

Comment:
Assert message here is misleading: it is more likely someone had called `grow` when they intended `maybe_grow`. See how it is done elsewhere:

```
void Node_Array::grow(uint i) {
  _nesting.check(_a); // Check if a potential reallocation in the arena is safe
  assert(i >= _max, "Should have been checked before, use maybe_grow?");
```

Reply:
Speaking of, we should probably move `_nesting.check(_a);` to `Node_Array::maybe_grow` as well.

Reply:
>  Assert message here is misleading

Yes, good point. I had basically reverted to before [JDK-8336999](https://bugs.openjdk.org/browse/JDK-8336999) but your assert message is better. Fixed.

> Speaking of, we should probably move _nesting.check(_a); to Node_Array::maybe_grow as well.

Right, I did that already. See changes in `src/hotspot/share/opto/node.hpp`.